### DISCUSSION TITLE
Am I using "engines" method on JSON API call right?

### DISCUSSION THREAD

**Discussion by prp-e** [searxng/searxng#3489] [Q&A] [answered]
This is how I use `engines` method/attribute in my code:

```
base_url = f"https://moa.mann-e.com/search?q={query}&format=json&categories=general&engines=google,bing"
``` 

And I am wondering if I'm doing it right or not. If not, how can I do it correctly?

**Comment by return42** [comment, marked as answer]
Seems you need to set `search_url`

https://github.com/searxng/searxng/blob/645a840d8211a016d808cf7da1ab62ff12d1db56/searx/engines/json_engine.py#L110

Have a look at settings.yml for mire examples ..

https://github.com/searxng/searxng/blob/645a840d8211a016d808cf7da1ab62ff12d1db56/searx/settings.yml#L1283-L1286
